src/WithdrawalManager.mo

```diff
@@ -86,6 +77,15 @@ module {
     /// creditAmount = amount of credit being deducted
     /// amount of tokens that the `to` account receives = creditAmount - userExpectedFee
     public func withdraw(p : ?Principal, to : ICRC1.Account, creditAmount : Nat, userExpectedFee : ?Nat) : async* WithdrawResponse {
+      let ok = switch(p) {
+        case null creditManager.burnPool(creditAmount);
+        case (?pp) creditManager.burn(pp, creditAmount);
+      };
+      if (not ok) {
+        let err = #InsufficientCredit;
+        log(ownPrincipal, #withdrawalError(err));
```

Comment:
I don't think we have to log this. We want to log all events that lead to any kind of state change. So that the state can be reproduced from the full log. This event isn't needed for that.